src/encoded/static/components/matrix_target.js

```diff
@@ -0,0 +1,677 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import pluralize from 'pluralize';
+import url from 'url';
+import PubSub from 'pubsub-js';
+import QueryString from '../libs/query_string';
+import { Panel, PanelBody, TabPanel, TabPanelPane } from '../libs/ui/panel';
+import { Modal, ModalHeader, ModalBody } from '../libs/ui/modal';
+import { svgIcon } from '../libs/svg-icons';
+import * as globals from './globals';
+import { MatrixInternalTags } from './objectutils';
+import { SearchFilter } from './matrix';
+import { TextFilter, SearchControls } from './search';
+import DataTable from './datatable';
+
+
+const VISUALIZE_LIMIT = 500;
+const SEARCH_PERFORMED_PUBSUB = 'searchPerformed';
+const SEARCH_RESULT_COUNT_PUBSUB = 'searchResultCount';
+const CLEAR_SEARCH_BOX_PUBSUB = 'clearSearchBox';
+
+const getTargetData = (context, assayTitle, organismName) => {
+    if (!context || !context.matrix || !context.matrix.x || !context.matrix.y || !assayTitle || !organismName) {
+        return null;
+    }
+
+    const subTabSource = 'biosample_ontology.classification';
+    const subTabList = context.matrix.x[subTabSource].buckets.map(x => x.key);
+    const targetData = {};
+
+    subTabList.forEach((subTab) => {
+        const xGroupBy1 = context.matrix.x.group_by[0];
+        const xGroupBy2 = context.matrix.x.group_by[1];
+        const xMatrix = context.matrix.x[xGroupBy1].buckets.find(f => f.key === subTab)[xGroupBy2]
+            .buckets
+            .reduce((a, b) => {
+                const m = a.concat(b);
+                return m;
+            }, [])
+            .map(x => x.key);
+        const xMatrixIndex = xMatrix.reduce((x, y, z) => { x[y] = z; return x; }, []);
+        const xMatrixLength = xMatrix.length;
+        const yGroupBy1 = context.matrix.y.group_by[0];
+        const yGroupBy2 = context.matrix.y.group_by[1];
+
+        const yData = context.matrix.y[yGroupBy1].buckets
+            .find(rBucket => rBucket.key === organismName)[yGroupBy2].buckets
+            .reduce((a, b) => {
+                const m = {};
+                m[b.key] = b[xGroupBy1].buckets
+                    .filter(f => f.key === subTab)
+                    .reduce((x, y) => {
+                        x.push(y[xGroupBy2].buckets
+                            .reduce((i, j) => i.concat(j), []));
+                        return x;
+                    }, []);
+                a.push(m);
+                return a;
+            }, []);
+
+        const yMatrixT = {};
+
+        yData.forEach((y) => {
+            const yKey = Object.keys(y)[0];
+            // IE11 does not support .fill. So .map is used.
+            yMatrixT[yKey] = yMatrixT[yKey] || [...Array(xMatrixLength + 1)].map(x => 0);
```

Comment:
You can use Array.prototype.fill on IE11 because babel polyfills it. In fact I'd recommend it here and the other place in here because they're causing ESLint errors.

Reply:
Done. I used ES5 (ecmascript 5) syntax because I think inline.js does not use babel.js to for transpiling.